### PR TITLE
Migrate from Slf4J to JCL. 

### LINKED ISSUE #2359: Migrate off SLF4J to Spring JCL
Spring Framework's JCL component provides a consistent logging layer that we should use instead of SLF4J.

See also spring-projects/spring-data-commons#2496

### PR DESCRIPTION
**Pull request by schauder**
Closes #2359

### CODE REVIEW

File: src/main/java/org/springframework/data/jpa/repository/query/NamedQuery.java [hunk 1]
```diff
@@ -134,15 +134,15 @@ public static RepositoryQuery lookupFrom(JpaQueryMethod method, EntityManager em
 
 		final String queryName = method.getNamedQueryName();
 
-		LOG.debug("Looking up named query {}", queryName);
+		LOG.debug(String.format("Looking up named query %s", queryName));
```
Comment (mp911de) [inline src/main/java/org/springframework/data/jpa/repository/query/NamedQuery.java:137]: Nit: Add `isDebugEnabled` guard to defer the cost of `String.format(…)`

File: src/main/java/org/springframework/data/jpa/repository/query/NamedQuery.java [hunk 2]
```diff
@@ -134,15 +134,15 @@ public static RepositoryQuery lookupFrom(JpaQueryMethod method, EntityManager em
...
 
 		if (!hasNamedQuery(em, queryName)) {
 			return null;
 		}
 
 		try {
 			RepositoryQuery query = new NamedQuery(method, em);
-			LOG.debug("Found named query {}!", queryName);
+			LOG.debug(String.format("Found named query %s!", queryName));
```
Comment (mp911de) [inline src/main/java/org/springframework/data/jpa/repository/query/NamedQuery.java:145]: Nit: Add `isDebugEnabled` guard to defer the cost of `String.format(…)`

File: src/main/java/org/springframework/data/jpa/support/MergingPersistenceUnitManager.java [hunk 3]
```diff
@@ -68,21 +68,21 @@ void postProcessPersistenceUnitInfo(MutablePersistenceUnitInfo pui, PersistenceU
 
 		for (URL url : oldPui.getJarFileUrls()) {
 			if (!pui.getJarFileUrls().contains(url)) {
-				LOG.debug("Adding JAR file URL {} to persistence unit {}.", url, persistenceUnitName);
+				LOG.debug(String.format("Adding JAR file URL %s to persistence unit %s.", url, persistenceUnitName));
```
Comment (mp911de) [inline src/main/java/org/springframework/data/jpa/support/MergingPersistenceUnitManager.java:71]: Nit: Add `isDebugEnabled` guard to defer the cost of `String.format(…)`

File: src/main/java/org/springframework/data/jpa/support/MergingPersistenceUnitManager.java [hunk 4]
```diff
@@ -68,21 +68,21 @@ void postProcessPersistenceUnitInfo(MutablePersistenceUnitInfo pui, PersistenceU
...
 				pui.addJarFileUrl(url);
 			}
 		}
 
 		for (String className : oldPui.getManagedClassNames()) {
 			if (!pui.getManagedClassNames().contains(className)) {
-				LOG.debug("Adding class {} to PersistenceUnit {}", className, persistenceUnitName);
+				LOG.debug(String.format("Adding class %s to PersistenceUnit %s", className, persistenceUnitName));
```
Comment (mp911de) [inline src/main/java/org/springframework/data/jpa/support/MergingPersistenceUnitManager.java:78]: Nit: Add `isDebugEnabled` guard to defer the cost of `String.format(…)`

File: src/main/java/org/springframework/data/jpa/support/MergingPersistenceUnitManager.java [hunk 5]
```diff
@@ -68,21 +68,21 @@ void postProcessPersistenceUnitInfo(MutablePersistenceUnitInfo pui, PersistenceU
...
 				pui.addManagedClassName(className);
 			}
 		}
 
 		for (String mappingFileName : oldPui.getMappingFileNames()) {
 			if (!pui.getMappingFileNames().contains(mappingFileName)) {
-				LOG.debug("Adding mapping file to persistence unit {}.", mappingFileName, persistenceUnitName);
+				LOG.debug(String.format("Adding mapping file to persistence unit %s.", mappingFileName, persistenceUnitName));
```
Comment (mp911de) [inline src/main/java/org/springframework/data/jpa/support/MergingPersistenceUnitManager.java:85]: Nit: Add `isDebugEnabled` guard to defer the cost of `String.format(…)`

**Review by mp911de — COMMENTED** [review]
A few places should be guarded with `isDebugEnabled`

**Comment by schauder** [comment]
I made the requested changes and merged the PR.

**Pull request closed without merging**